build-gluster-org/scripts/generic_package.sh

```diff
@@ -0,0 +1,200 @@
+
+# Check for OS(Ubuntu or Debian)
+if [ "$os" == "ubuntu" ]; then
+        mirror="http://ubuntu.osuosl.org/ubuntu/"
+	debuild_key=4F5B5CA5
+elif [ "$os" == "debian" ]; then
+        mirror="http://ftp.us.debian.org/debian/"
+	case ${series} in
+	  "3.12")
+    	    debuild_key=${debuild_keys[0]}
+    	    pbuild_key=${pbuild_keys[0]}
+    	    ;;
+  	  "4.0")
+            debuild_key=${debuild_keys[1]}
+            pbuild_key=${pbuild_keys[1]}
+            ;;
+          "4.1")
+            if [ "$flavor" == "stretch" ]; then
+                 debuild_key=${pbuild_keys[2]}
+            else
+                 debuild_key=${debuild_keys[2]}
+            fi
+            pbuild_key=${pbuild_keys[2]}
+            ;;
+          "5" | "6" | "7")
+            if [ "$flavor" == "stretch" ]; then
+                 debuild_key=${pbuild_keys[3]}
+            else
+                 debuild_key=${debuild_keys[3]}
+            fi
+            pbuild_key=${pbuild_keys[3]}
+        esac
+else
+	echo "Exiting: OS should be debian or ubuntu. Please provide the right one"
+	exit
+fi
+
+mkdir ${os}-${flavor}-Glusterfs-${version}
+
+cd ${os}-${flavor}-Glusterfs-${version}
+
+mkdir build packages
+
+echo "Building glusterfs-${version}-${release} for ${flavor}"
+
+cd build
+
+TGZS=(`ls ~/glusterfs-${version}-?-*/build/glusterfs-${version}.tar.gz`)
+echo ${TGZS[0]}
+
+if [ -z ${TGZS[0]} ]; then
+        echo "wget https://download.gluster.org/pub/gluster/glusterfs/${series}/${version}/glusterfs-${version}.tar.gz"
+        wget https://download.gluster.org/pub/gluster/glusterfs/${series}/${version}/glusterfs-${version}.tar.gz
+else
+        echo "found ${TGZS[0]}, using it..."
+        cp ${TGZS[0]} .
+fi
+
+echo "Creating link file.."
+ln -s glusterfs-${version}.tar.gz glusterfs_${version}.orig.tar.gz
+
+echo "Untaring.."
+tar xpf glusterfs-${version}.tar.gz
+
+# Changelogs needed for building are maintained in a separate repo.
+# the repo has to be clone and updated properly so we can copy the changelogs so far.
+
+echo "Cloning the glusterfs-debian repo"
+git clone http://github.com/gluster/glusterfs-debian.git
+
+cd glusterfs-debian/
+
+git checkout -b ${flavor}-${series}-local origin/${flavor}-glusterfs-${series}
+
+if [ "$os" == "ubuntu" ]; then
+	sed -i "1i glusterfs (${version}-${os}1~${flavor}1) ${flavor}; urgency=medium\n\n * GlusterFS ${version} GA\n\n -- GlusterFS GlusterFS deb packages <deb.packages@gluster.org>  `date +"%a, %d %b %Y %T %z"`\n" debian/changelog
+elif [ "$os" == "debian" ]; then
+	sed -i "1i glusterfs (${version}-1) ${flavor}; urgency=low\n\n * GlusterFS ${version} GA\n\n -- Gluster Packager <glusterpackager@download.gluster.org>  `date +"%a, %d %b %Y %T %z"`\n" debian/changelog
+fi
+
+git commit -a -m "Glusterfs ${version} G.A (${flavor})"
+
+echo "Copying Changelog to source"
+cp -a debian ../glusterfs-${version}/
+
+echo "Building source package.."
+cd ../glusterfs-${version}
+debuild -S -sa -k${debuild_key}
```

Comment:
Where would the signing key stored ? (since -k requires that)

Reply:
debuild_key variable contains the key.

Reply:
I wasn't clear on my question. From what I understand, -k is the key identifier. But the actual private key is somewhere else (I think in ~/.gnupg), and so we need to discuss how that part is going to be managed.

Reply:
we are initialising the values for debuild keys in https://github.com/gluster/build-jobs/pull/44/files/b32e0625d90b8c89432e753aab502d5e72b64b46#diff-02d2c32f5282eea2a7412f831420ae8c2f55ce0ed17671f6878e110c258b2e6bR29

The key creation is something we have maintained in a particular machine, whose access is not given to wider audience.

Reply:
Then, where is that machine ?

Reply:
Here is the draft of mojo doc that contains machine details: https://source.redhat.com/.motion/communitiesatredhat/crosscuttingco/pjmcop/prodev/study_grp/study_group_sign_up/A883E22D-0D6C-EB11-80F2-000D3A020FEB/.object

Reply:
Page does not exist :/

Reply:
Please checkout this link: https://source.redhat.com/.motion/communitiesatredhat/crosscuttingco/pjmcop/pjm_prodev/pjm_study_grp/pjm_study_group_sign_up/a883e22d-0d6c-eb11-80f2-000d3a020feb/.object